examples/MultipleRedeemers.purs

```diff
@@ -0,0 +1,309 @@
+      (\(tn /\ n) -> Value.singleton cs tn $ fromInt n)
+      t
+
+    values :: Array (Value /\ Redeemer)
+    values = lmap (toCsValue toks) <$> css
+
+    vhashes :: Array ValidatorHash
+    vhashes = validatorHash <<< fst <$> validators
+
+  logInfo' $ "Trying to create " <> show values
+
+  let
+    lookups :: Lookups.ScriptLookups PlutusData
+    lookups = mconcat
+      [ mconcat $ Lookups.validator <<< fst <$> validators
+      , mconcat $ Lookups.mintingPolicy <<< fst <$> policies
+      ]
+
+    constraints :: Constraints.TxConstraints Unit Unit
+    constraints = mconcat
+      [ mconcat $ do
+          _ <- vhashes
+          val /\ red <- values
+          pure $ Constraints.mustMintValueWithRedeemer red val
+      -- There will be `amount` UTxOs for every `CurrencySymbol`, `TokenName` and
+      -- `Validator. Every UTxO will contain exactly 1 token.
+      , mconcat $ do
+          vhash <- vhashes
+          (cs /\ _) <- css
+          (tok /\ amount) <- toks
+          replicate amount
+            $ Constraints.mustPayToScript vhash unitDatum DatumWitness
+            $ flip toCsValue cs
+            $ pure
+            $ Tuple tok 1
+      , Constraints.mustIncludeDatum unitDatum
+      ]
+
+  unbalancedTx <- liftedE $ Lookups.mkUnbalancedTx lookups constraints
+  balancedSignedTx <- liftedE $ balanceAndSignTxE unbalancedTx
+  logInfo' $ "Balanced and signed tx is " <> show balancedSignedTx
+  txId <- submit balancedSignedTx
+  logInfo' $ "Tx ID: " <> show txId
+  pure txId
+
+-- | for each Script we have one redeemer that we're goig to supply
+-- | for each MintingPolicy we spend the specified count of tokens with the names specified
+-- | but one token each. We do not add datums, the redeemers are tupled up with the Scripts
+spendTokens
+  :: Configuration -> TransactionHash -> Contract () Unit
+spendTokens cfg@{ validators, policies } hash = do
+  utxosnreds :: Array (UtxoMap /\ Redeemer) <- getUtxos cfg hash
+
+  self :: PaymentPubKeyHash <- liftedM "Could not obtain own PaymentPubKeyHash"
+    ownPaymentPubKeyHash
+
+  logInfo' $ "Found " <> show (getorefs hash <<< fst <$> utxosnreds) <>
+    " at validators' addresses"
+
+  let
+    lookups :: Lookups.ScriptLookups PlutusData
+    lookups = mconcat
+      [ mconcat $ Lookups.unspentOutputs <<< fst <$> utxosnreds
+      , mconcat $ Lookups.validator <<< fst <$> validators
+      , mconcat $ Lookups.mintingPolicy <<< fst <$> policies
+      , Lookups.ownPaymentPubKeyHash self
+      ]
+
+    constraints :: Constraints.TxConstraints Unit Unit
+    constraints = mconcat $
+      [ mconcat $ do
+          (utxo /\ red) <- utxosnreds
+          flip Constraints.mustSpendScriptOutput red <$> getorefs hash utxo
+      ]
+
+  ubTx <- liftedE $ Lookups.mkUnbalancedTx lookups constraints
+  bsTx <-
+    liftedM "Failed to balance/sign tx" $ balanceAndSignTx ubTx
+  hash2 <- submit bsTx
+  logInfo' $ "Hash of second transaction " <> show hash2
+  pure unit
+
+getUtxos
```

Comment:
Is there a reason why we don't `awaitTxConfirmed` the `hash` first, rather than looping until success?

Reply:
Simply because this function didn't exist back when I wrote this.